File: frontend/src/models/building.ts
import {BuildingType} from "./buildingType";
import {TileIdentifier} from "./tile";
import {DetailLogEntry} from "./detailLogEntry";

export interface Building {
    type: BuildingType,
    active: boolean,
    tile: TileIdentifier | null,
    details: DetailLogEntry<BuildingDetailType>[]
}

export type  BuildingDetailType
    = "WORKED_TILE"
    | "ACTIVITY"
    | "CONSUMED"
    | "PRODUCED"
    | "MISSING"

File: backend/strategy-game-common-data/src/main/kotlin/io/github/smiley4/strategygame/backend/commondata/Building.kt
package io.github.smiley4.strategygame.backend.commondata


class Building(
    val type: BuildingType,
    var workedTile: TileRef?,
    var validity: BuildingValidity,
    val activity: BuildingActivity
)

class BuildingValidity(
    var workTile: Boolean,
    var inputResources: Boolean,
)

class BuildingActivity(
    val consumed: ResourceCollection,
    val produced: ResourceCollection,
    val missing: ResourceCollection
)